zerver/lib/generate_test_data.py

```diff
@@ -135,7 +138,19 @@ def add_flair(paragraphs: List[str], gens: Dict[str, Any]) -> List[str]:
             # even when developing offline).
             txt = paragraphs[i] + "\n" + next(gens["images"])
 
-        results.append(txt)
+        # Attachments:
+        # Similar to images, we can use files already in the project as
+        # attachments. However, we must hit the `upload` endpoint, which
+        # has to be performed in populate_db.py. Therefore, the attachment
+        # path is not included as part of the message text but appended
+        # after the text.
+        if txt is not None:
+            txt_with_attachment_path = [txt]
+        elif key == "attachment":
+            txt_with_attachment_path = [paragraphs[i]]
+            txt_with_attachment_path.append(next(gens["attachment-paths"]))
```

Comment:
`[text, attachment, ..., attachment]` is a confusing use of the list type. It’d be clearer to make a class instead:

```python
@dataclass
class TestMessage:
    text: str
    attachment_paths: List[str]
```

Reply:
That's a great idea. However, I think I should get my current code to work (to pass tests) before I can adopt it in a following commit.

Reply:
I just adopted your idea.